client/main.lua

```diff
@@ -1,583 +1,562 @@
+        
+        if items_attatched[item].slot then
+            PlayerSlots[items_attatched[item].tier][items_attatched[item].slot].isBusy = false
+        end
+        
+        items_attatched[item] = nil
+    end
+    
+    WeapDelete = false
 end
 
+
 -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-local carryingBox = nil
+--** CARRYING FUNCTIONS **--
 
+local carryingBox = nil
 local function requestAnimDict(animDict)
-	if not HasAnimDictLoaded(animDict) then
-		RequestAnimDict(animDict)
-
-		while not HasAnimDictLoaded(animDict) do
-			Wait(0)
-		end
-	end
+    if not HasAnimDictLoaded(animDict) then
+        RequestAnimDict(animDict)
+        
+        while not HasAnimDictLoaded(animDict) do
+            Wait(0)
+        end
+    end
 end
 
-
--- Should probably allow for different animations for different items down the line --
 local function doAnim(item)
-  if carryingBox then return end -- Only allow the function to be run once at a time
-  carryingBox = item
-  local ped = PlayerPedId()
-  local dict, anim = props[items].dict or 'anim@heists@box_carry@', props[item].anim or 'idle'
-  if anim == 'none' then return end
-
-  requestAnimDict(dict)
-  CreateThread(function()
-    while carryingBox do
-      if not IsEntityPlayingAnim(ped, dict, anim, 3) then
-        TaskPlayAnim(ped, dict, anim, 8.0, -8, -1, 49, 0, 0, 0, 0)
-      end
-
-      if props[carryingBox].blockAttack then
-        DisableControlAction(0, 24, true) -- disable attack
-        DisableControlAction(0, 25, true) -- disable aim
-        DisableControlAction(0, 47, true) -- disable weapon
-        DisableControlAction(0, 58, true) -- disable weapon
-        DisableControlAction(0, 263, true) -- disable melee
-        DisableControlAction(0, 264, true) -- disable melee
-        DisableControlAction(0, 257, true) -- disable melee
-        DisableControlAction(0, 140, true) -- disable melee
-        DisableControlAction(0, 141, true) -- disable melee
-        DisableControlAction(0, 142, true) -- disable melee
-        DisableControlAction(0, 143, true) -- disable melee
-      end
-
-      if props[carryingBox].blockCar and IsPedGettingIntoAVehicle(ped) then
-        ClearPedTasksImmediately(ped) -- Stops all tasks for the ped
-      end
-
-      if props[carryingBox].blockRun then
-        DisableControlAction(0, 22, true) -- disable jumping
-        DisableControlAction(0, 21, true) -- disable sprinting
-      end
-
-      Wait(1)
-    end
-
-    ClearPedTasks(ped)
-  end)
-end
+    if carryingBox then return end -- Only allow the function to be run once at a time
+    carryingBox = item
+    local ped = PlayerPedId()
+    local dict, anim = props[item].dict or 'anim@heists@box_carry@', props[item].anim or 'idle'
+    if anim == 'none' then return end

```

Comment:
Only conflict I can really see is that you didnt change the if statement in doAnim to ` if not anim or not dict then return end`

Reply:
oh neverminded that wouldnt make sense, not sure 😂